core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.controller.ts

```diff
@@ -120,122 +156,147 @@ angular.module('oppia').directive('topicsAndSkillsDashboardPage', [
-                    ImageLocalStorageService.flushStoredImagesData();
-                    SkillCreationService.resetSkillDescriptionStatusMarker();
-                    $uibModalInstance.dismiss('cancel');
-                  };
-                }
-              ]
+              controller: 'CreateNewSkillModalController'
             }).result.then(function(result) {
               ContextService.resetImageSaveDestination();
               SkillCreationService.createNewSkill(
                 result.description, result.rubrics, result.explanation, []);
             });
           };
+          /**
+           * @param {Number} pageNumber - Page number to navigate to.
+           */
+          ctrl.goToPageNumber = function(pageNumber) {
+            if (ctrl.activeTab === ctrl.TAB_NAME_TOPICS) {
+              ctrl.topicPageNumber = pageNumber;
+              ctrl.pageNumber = ctrl.topicPageNumber;
+              ctrl.displayedTopicSummaries =
+                  ctrl.topicSummaries.slice(
+                    pageNumber * ctrl.itemsPerPage,
+                    (pageNumber + 1) * ctrl.itemsPerPage);
+            } else if (ctrl.activeTab === ctrl.TAB_NAME_UNTRIAGED_SKILLS) {
+              ctrl.totalEntityCountToDisplay =
+                  ctrl.totalUntriagedSkillSummaries.length;
+              ctrl.skillPageNumber = pageNumber;
+              ctrl.pageNumber = ctrl.skillPageNumber;
+              ctrl.untriagedSkillSummaries =
+                  ctrl.totalUntriagedSkillSummaries.slice(
+                    pageNumber * ctrl.itemsPerPage,
+                    (pageNumber + 1) * ctrl.itemsPerPage);
+            }
+          };
+          /**
+           * @param {String} direction - Direction, whether to change the
+           * page to left or right by 1.
+           */
+          ctrl.changePageByOne = function(direction) {
+            ctrl.lastPage = parseInt(
+              String(ctrl.totalEntityCountToDisplay / ctrl.itemsPerPage));
+            if (direction === ctrl.MOVE_TO_PREV_PAGE && ctrl.pageNumber >= 1) {
+              ctrl.goToPageNumber(ctrl.pageNumber - 1);
+            } else if (direction === ctrl.MOVE_TO_NEXT_PAGE &&
+                ctrl.pageNumber < ctrl.lastPage) {
+              ctrl.goToPageNumber(ctrl.pageNumber + 1);
+            }
+          };
+
+          ctrl.applyFilters = function() {
+            ctrl.topicSummaries = (
+              TopicsAndSkillsDashboardPageService.getFilteredTopics(
+                ctrl.totalTopicSummaries, ctrl.filterObject));
+
+            ctrl.displayedTopicSummaries =
+                ctrl.topicSummaries.slice(0, ctrl.itemsPerPage);
+            ctrl.currentCount = ctrl.topicSummaries.length;
+            ctrl.goToPageNumber(0);
+            $scope.$applyAsync();
+            _forceSelect2Refresh();
+          };
+
+          var _forceSelect2Refresh = function() {
+            ctrl.select2DropdownIsShown = false;
+            $timeout(function() {
```

Comment:
Why do we need this? Maybe add a comment.

Reply:
Done.